### PR TITLE
Fix rbac 3308

### PR DESCRIPTION
**Pull request by dereknola**
#### Proposed Changes ####
Change the name of the default `cloud-controller-manager` ClusterRole and ClusterRoleBindings to `k3s-cloud-controller-manager` to prevent conflicts with external cloud providers.

#### Types of Changes ####
Bugfix

#### Verification ####
`k3s kubectl get clusterrolebinding` should now output `k3s-cloud-controller-manager` as a ClusterRole.

#### Linked Issues ####
https://github.com/k3s-io/k3s/issues/3308

#### Further Comments ####
This is my first PR for k3s, so if there is something wrong with how I filled out this request, or some other information I should be adding, please let me know.

### CODE REVIEW

File: pkg/daemons/control/server.go
Comment (brandond) [inline pkg/daemons/control/server.go:430]: We try to use version.Program or version.ProgramUpper instead of using the `k3s` string literal, as this white-labeling allows rke2 to not have references to k3s all over the place. We aren't quite 100% on this yet, but that's the convention we're going for.

```suggestion
	_, err = crb.Get(version.Program + "-cloud-controller-manager", metav1.GetOptions{})
```

**Review by brandond — CHANGES_REQUESTED** [review]

File: manifests/ccm.yaml
Comment (brandond) [inline manifests/ccm.yaml:79]: This needs to be changed too, along with updating the cloud-controller-manager certificate subject here:
https://github.com/k3s-io/k3s/blob/master/pkg/daemons/control/deps/deps.go#L297

**Review by brandond — CHANGES_REQUESTED** [review]

File: manifests/ccm.yaml
Comment (dereknola) [reply manifests/ccm.yaml:79]: Changed in new commit.

File: pkg/daemons/control/deps/deps.go
Comment (galal-hussein) [inline pkg/daemons/control/deps/deps.go:297]: that will create a conflict with rke2 since in the manifest is statically set to k3s- not version.program, for reference check how we put a template variable to the manifests https://github.com/k3s-io/k3s/blob/master/pkg/server/server.go#L230 https://github.com/k3s-io/k3s/blob/master/manifests/coredns.yaml

**Review by galal-hussein — CHANGES_REQUESTED** [review]

File: manifests/ccm.yaml
Comment (galal-hussein) [reply manifests/ccm.yaml:79]: https://github.com/k3s-io/k3s/pull/3388#pullrequestreview-674630108, this should fix this problem

File: pkg/daemons/control/deps/deps.go
Comment (brandond) [reply pkg/daemons/control/deps/deps.go:297]: @galal-hussein yes, it is my expectation that RKE2 will need to be updated as well once this is pulled through. RKE2 does not use the ccm.yaml manifest that's embedded in k3s, instead the roles are created via bootstrap code at https://github.com/rancher/rke2/blob/master/pkg/rke2/clusterrole_templates.go#L6

File: manifests/ccm.yaml
Comment (brandond) [reply manifests/ccm.yaml:79]: The manifest can have `k3s` embedded in it, since RKE2 does not use any of the package manifests (they are disabled via build flag).

**Review by brandond — APPROVED** [review]

**Comment by brandond** [comment]
LGTM. Make sure you squash before merging.

**Comment by briandowns** [comment]
Did this need another approval?

**Pull request merged**
